### PR TITLE
Added documentation about autoscaling

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: content/docs/next-release/operator.md [hunk 1]
```diff
@@ -147,13 +147,30 @@ The available strategies are described in the following sections.
...
 
-The agent can be injected as a sidecar on the instrumented application or as a daemonset.
+The agent can be injected as a sidecar on the instrumented application or as a daemonset. 
+
+The collector can be configured to autoscale on demand. By default, when no `.Spec.Collector.Replicas` is provided, the Jaeger Operator will create a Horizontal Pod Autoscaler (HPA) configuration for the collector. We recomment setting an explicit value for `.Spec.Collector.MaxReplicas`, along with reasonable value for the resources that the collector's pod is expected to consume. When no `.Spec.Collector.MaxReplicas` is set, the operator will set `100` as its value. Read more about HPA on [Kubernetes' website](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/). The feature can be explicitly disabled by setting `.Spec.Collector.Autoscale` to `false`. Here's an example, setting the collector's limits as well as the maximum number of replicas:
```
Comment (JStickler) [inline content/docs/next-release/operator.md]: By default, when no value for `.Spec.Collector.Replicas` is provided, the Jaeger Operator will create a Horizontal Pod Autoscaler (HPA) configuration for the collector.

Comment (JStickler) [inline content/docs/next-release/operator.md]: We recommend setting an explicit value for `.Spec.Collector.MaxReplicas`, along with a reasonable value for the resources that the collector's pod is expected to consume.

**Comment by jpkrohling** [comment]
@JStickler could you please re-review? Once I get your approval, I'll merge it.

**Comment by JStickler** [comment]
LGTM

**Pull request merged**